activerecord/lib/active_record/relation/query_methods.rb

```diff
@@ -110,6 +110,8 @@ def missing(*associations)
           @scope.left_outer_joins!(association)
           if @scope.table_name == reflection.table_name
             @scope.where!(association => { reflection.association_primary_key => nil })
+          elsif (@scope.values[:joins] && @scope.values[:left_outer_joins]) && ((@scope.values[:joins] & @scope.values[:left_outer_joins]).size != @scope.values[:joins].size)
```

Comment:
Hey @paulreece thanks for the PR.

I don't think this is the correct fix for the problem. We shouldn't need to reach into the scope values like this and make such a complex elsif statement. I'm unsure if the original fix was correct either, and don't really have the bandwidth to dig in and find the right fix. @fatkodima since you made the original changes, maybe you have an idea of what a better approach is?

Also in the future, the contributors [discord](https://discord.gg/HGRyZPXEjF) is a better place to ask for help with a PR. 😄

Reply:
To be a bit clearer - I think comparing `@scope.table_name == reflection.table_name` in the original fix isn't correct because here we have a case where the table names don't match but we want the `association` as the key in the where. So we need to figure out how to know when we want an association instead of the reflection's table name.

Reply:
I really appreciate the feedback, apologies for pinging you here and not on Discord.  I will definitely follow the Discord protocol in the future.  I'll experiment with trying to get this to work from the method setup before the original fix.

Reply:
Okay, after comparing the different error cases tonight. I think I found a way simpler solution.  After deep diving into for a while I realized that it seems like we only want to use  `reflection.table_name` when the `@scope.values` hash only contains one key in `:left_outer_joins`.  This covers the problem found in [47909](https://github.com/rails/rails/issues/47909).  Then if the `@scope.values` hash has more than one key we use the `association`.  This works for both [44964](https://github.com/rails/rails/issues/44964) _and_ the newest issue this PR was submitted to address [48651](https://github.com/rails/rails/issues/48651).

The current suggested change is now:
```Ruby
if @scope.values.size > 1
  @scope.where!(association => { reflection.association_primary_key => nil })
 else
  @scope.where!(reflection.table_name => { reflection.association_primary_key => nil })
end
```

It passes the all the test cases for ActiveRecord including a new one I added for the newest issue [48651](https://github.com/rails/rails/issues/48651).